data/success_stories/ahrefs-full-stack-web.md

```diff
@@ -0,0 +1,72 @@
+---
+title: Full-Stack OCaml Web Application With React
+logo: success-stories/ahrefs.svg
+card_logo: success-stories/white/ahrefs.svg
+background: /success-stories/ahrefs-full-stack-bg.jpg
+theme: blue
+synopsis: "Ahrefs transitioned from PHP/jQuery to full-stack OCaml using Melange and React, eliminating team silos and enabling any engineer to contribute across their entire web application stack."
+url: https://ahrefs.com/
+priority: 2
+why_ocaml_reasons:
+- Type Safety
+- Unified Technology Stack
+- Team Efficiency
+- Integration with JavaScript Ecosystem
+- Shared Data Types
+- Developer Productivity
+- Code Reliability
+---
+
+## Challenge
+
+[Ahrefs](https://ahrefs.com/) is a Singapore-based SaaS company that provides SEO tools and marketing intelligence powered by big data. Since 2011, they've built their business around OCaml, using it for web crawling and data processing to serve thousands of customers worldwide. Today, they're trusted by 44% of Fortune 500 companies and operate as a lean, self-funded organization focused on efficiency.
+
+By 2017, Ahrefs had built a successful SEO tools business powered by OCaml on the backend, but they faced a bottleneck in web application development. Their frontend was built with PHP and jQuery while their data processing lived in OCaml. Every time frontend developers needed backend data, they had to coordinate with backend engineers to update the APIs.
+
+Ahrefs wanted engineers to be productive across the entire stack, but the technology divide made this unnecessarily difficult. The JavaScript tooling used in 2017 for the frontend of the web application was lacking compared to today's TypeScript ecosystem. Ahrefs had already built years of expertise in OCaml. The question became: could they extend OCaml's benefits to the frontend?
+
+The challenge was both technical and cultural. Could they transition the entire frontend team to a OCaml? Even when some of the engineers hadn't used a functional programming language before? Would the benefits of a unified stack outweigh the costs?
+
+## Result
+
+After adopting Reason/BuckleScript around 2017-2020 and migrating to **[Melange](https://melange.re/)** in 2023, Ahrefs has achieved a full stack web development setup around OCaml.
+
+Now, any engineer in the company can contribute across the entire web application. Thanks to shared types between backend and frontend, coordination overhead is greatly reduced. 
+
+Frontend and backend stay in sync: When data structures change, type errors guide developers to update all affected code.
+
+Today, their **5 billion daily frontend requests** are handled by the same OCaml codebase that powers their backend systems. The web application serving **44% of Fortune 500 companies** is built from **1.5 million lines of OCaml code** spanning both frontend and backend.
+
+## Why OCaml
+
+For Ahrefs, extending OCaml to the frontend wasn't about technological purity—it was about simplifying their business.
+
```

Comment:
I would mention the "mental model" of one language, and remove the "sharing types" since it has been repeated above